tests/unit/Makefile

```diff
@@ -7,62 +27,55 @@ LIBGRAPHBLAS=../../deps/GraphBLAS/build/libgraphblas.a
+run_tests: $(TESTS)
+	@(sh -c ./value_handler)
 
-tuples_iter: test_tuples_iter.o
-	$(CC) $(CFLAGS) -o test_tuples_iter test_tuples_iter.o  $(DEPS) $(LDFLAGS)
+all : $(TESTS) run_tests
 
-filter_tree: test_filter_tree.o
-	$(CC) $(CFLAGS) -o test_filter_tree test_filter_tree.o  $(DEPS) $(LDFLAGS)
+clean :
+	rm -f $(TESTS) gtest.a gtest_main.a *.o
 
-value: test_value.o
-	$(CC) $(CFLAGS)  -o test_value test_value.o  $(DEPS) $(LDFLAGS)
+# Builds gtest.a and gtest_main.a.
 
-arithmetic_expression: test_arithmetic_expression.o
-	$(CC) $(CFLAGS)  -o test_arithmetic_expression test_arithmetic_expression.o  $(DEPS) $(LDFLAGS)
+# Usually you shouldn't tweak such internal variables, indicated by a
+# trailing _.
+GTEST_SRCS_ = $(GTEST_DIR)/src/*.cc $(GTEST_DIR)/src/*.h $(GTEST_HEADERS)
 
-resultset_record: test_resultset_record.o
-	$(CC) $(CFLAGS)  -o test_resultset_record test_resultset_record.o  $(DEPS) $(LDFLAGS)
+# For simplicity and to avoid depending on Google Test's
+# implementation details, the dependencies specified below are
+# conservative and not optimized.  This is fine as Google Test
+# compiles fast and for ordinary users its source rarely changes.
+gtest-all.o : $(GTEST_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+            $(GTEST_DIR)/src/gtest-all.cc
 
-build: graph algebraic_expression tuples_iter filter_tree value arithmetic_expression resultset_record
+gtest_main.o : $(GTEST_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+            $(GTEST_DIR)/src/gtest_main.cc
 
-test_graph:
-	@(sh -c ./test_graph)
-.PHONY: test_graph
+gtest.a : gtest-all.o
+	$(AR) $(ARFLAGS) $@ $^
 
-test_algebraic_expression:
-	@(sh -c ./test_algebraic_expression)
-.PHONY: test_algebraic_expression
+gtest_main.a : gtest-all.o gtest_main.o
+	$(AR) $(ARFLAGS) $@ $^
 
-test_tuples_iter:
-	@(sh -c ./test_tuples_iter)
-.PHONY: test_tuples_iter
+# A test should link with either gtest.a or gtest_main.a, depending on whether it defines its own main()
+# function.
 
-test_filter_tree:
-	@(sh -c ./test_filter_tree)
-.PHONY: test_filter_tree
+SRCDIR := $(shell pwd)
+LDFLAGS :=  -lc -lm -ldl -lpthread
 
-test_value:
-	@(sh -c ./test_value)
-.PHONY: test_value
+.SUFFIXES: .cc .c .so .xo .o
 
-test_arithmetic_expression:
-	@(sh -c ./test_arithmetic_expression)
-.PHONY: test_arithmetic_expression
+test_value.o : test_value.c
+	$(CC) $(CFLAGS) -c test_value.c $(LDFLAGS)
 
-test_resultset_record:
-	@(sh -c ./test_resultset_record)
-.PHONY: test_resultset_record
+value_handler.o : value_handler.cc $(GTEST_HEADERS)
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(CXXSUPPRESS) -c value_handler.cc $(LDFLAGS)
 
-test: test_graph test_algebraic_expression test_tuples_iter test_filter_tree test_value test_arithmetic_expression test_resultset_record
+value_handler: ../../src/value.o test_value.o value_handler.o gtest_main.a
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(CXXSUPPRESS) -lpthread $^ -o $@
```

Comment:
A C++ handler file is built and directly invokes the C functions from our `test_value.o` file.